judge/judgedaemon.main.php

```diff
@@ -241,12 +241,13 @@ function read_judgehostlog(int $n = 20) : string
 function fetch_executable(
     string $workdirpath, string $execid, string $md5sum, bool $combined_run_compare = false) : array
 {
-    $execpath = "$workdirpath/executable/" . $execid;
+    $execpath = "$workdirpath/executable/download/" . $execid;
     $execmd5path = $execpath . "/md5sum";
-    $execdeploypath = $execpath . "/.deployed";
-    $execbuildpath = $execpath . "/build";
-    $execrunpath = $execpath . "/run";
     $execzippath = $execpath . "/executable.zip";
+    $execdeploypath = $execpath . "/.deployed";
+    $execbuilddir = "$workdirpath/executable/build/" . $execid;
+    $execbuildpath = $execbuilddir . "/build";
+    $execrunpath = $execbuilddir . "/run";
```

Comment:
I propose the change the directory structure a bit: make subdirectories `download` and `build` inside a directory `$workdirpath/executable/$execid/`. I think that's a more natural grouping.
Secondly, this makes it easier to make the `build` directory a subdirectory of the chroot, which means that that `build` directory is now clean, and doesn't contain the `etc`, `usr`, `proc`, etc. directories from the chroot bind mounts, which could potentially cause conflicts.